src/Couchbase.Lite.Shared/API/Query/IndexBuilder.cs

```diff
@@ -27,20 +28,28 @@ namespace Couchbase.Lite.Query
     /// </summary>
     public static partial class IndexBuilder
     {
+        #region Constants
+
+        private const string Tag = nameof(IndexBuilder);
+
+        #endregion
+
         /// <summary>
         /// Starts the creation of an index based on a simple property
         /// </summary>
         /// <param name="items">The items to use to create the index</param>
         /// <returns>The beginning of a value based index</returns>
         [NotNull]
-        public static IValueIndex ValueIndex(params IValueIndexItem[] items) => new QueryIndex(items);
+        public static IValueIndex ValueIndex([NotNull]params IValueIndexItem[] items) => 
+            new QueryIndex(CBDebug.MustNotBeNull(WriteLog.To.Query, Tag, nameof(items), items));
```

Comment:
This won't work because I don't think `params` can ever be null.  There needs to be a loop that checks each item for `null` instead.  Consider enhancing `CBDebug` with something like `MustNotContainNull`.  Check with @pasin to see if that is what he is planning to do or if he is just checking for `null` collections.

Reply:
Base on the chat I had with @jayahariv , iOS checks null for the collections for ValueIndex at this moment.

Reply:
`ItemNotNull` means that no items contained in the collection should be null.  To guarantee that you can't just do `CBDebug.MustNotBeNull`.  You have to check each item for null like I mentioned above.  If we are not going to check that then this parameter should have no attribute.